web-ui/src/main/resources/catalog/components/edit/onlinesrc/OnlineSrcService.js

```diff
@@ -367,7 +367,7 @@
             uuidref: qParams.uuidDS,
             uuid: qParams.uuidSrv,
             process: qParams.process
-          }).then(function() {
+          }, true).then(function() {
```

Comment:
this should be false no ? You want to save the current editor states, then add service info in dataset, then add dataset info in service.

Reply:
I will recheck, but I tested linking a service to a dataset that is this code and didn't notice issues, but was not adding additional changes in the editor.

Reply:
@fxprunayre I have recheck this and when removing the `true` parameter it happens the error reported, seem it's required in both `linkToDataset` and `linkToService`, but if you have time to test would be great. 

Not sure if many users use the associated resources panel with the INSPIRE view (or other view using template fields), but in such a case, it's not nice to get the metadata corrupted. Any additional tests for the fix is welcome.

Reply:
Ok if you've checked that the current user inputs are properly saved first before running any processes then it is good.

Reply:
> it's not nice to get the metadata corrupted.

This one and the deadlock issues are maybe 2 good reasons for releasing 3.4.2 ?

Reply:
Yes, but I would like to fix also https://github.com/geonetwork/core-geonetwork/issues/2582. Lets plan the release for early next week, so I can manage about this other issue.